code/core/src/phone/java/com/adobe/marketing/mobile/services/ui/MessageFragment.java

```diff
@@ -172,7 +174,68 @@ public void onCreate(final Bundle savedInstanceState) {
                                 .getApplicationContext(),
                         webViewGestureListener);
 
-        setStyle(DialogFragment.STYLE_NORMAL, android.R.style.Theme_Translucent_NoTitleBar);
+        setStyle(DialogFragment.STYLE_NO_FRAME, android.R.style.Theme_Translucent_NoTitleBar);
+    }
+
+    @Override
+    public Dialog onCreateDialog(final Bundle savedInstanceState) {
+        final Activity parentActivity = getActivity();
+        final int fragmentTheme = getTheme();
+        final Dialog defaultDialog = super.onCreateDialog(savedInstanceState);
+
+        if (parentActivity == null || message == null) {
+            Log.trace(
+                    ServiceConstants.LOG_TAG,
+                    TAG,
+                    "%s (%s), returning a default Dialog object.",
+                    parentActivity == null ? "Parent Activity" : "Message",
+                    UNEXPECTED_NULL_VALUE);
+            return defaultDialog;
+        }
+
+        final MessageSettings messageSettings = message.getMessageSettings();
+        if (messageSettings == null) {
+            Log.trace(
+                    ServiceConstants.LOG_TAG,
+                    TAG,
+                    "%s (MessageSettings), returning a default Dialog object.",
+                    UNEXPECTED_NULL_VALUE);
+            return defaultDialog;
+        }
+
+        final boolean uiTakeoverEnabled = messageSettings.getUITakeover();
+        return new Dialog(parentActivity, fragmentTheme) {
+            @Override
+            public boolean onTouchEvent(@NonNull final MotionEvent motionEvent) {
+                if (!uiTakeoverEnabled) {
+                    // only log action up or down events as this logging can get quite
+                    // spammy
+                    if (motionEvent.getAction() == MotionEvent.ACTION_DOWN
```

Comment:
Since each action down is paired with an action up ultimately - you can just log when the type is `MotionEvent.ACTION_UP`.